e2e/test/iothub/service/FileUploadNotificationE2ETest.cs

```diff
@@ -18,41 +16,124 @@
+        [DataRow(TransportType.Amqp)]
+        [DataRow(TransportType.Amqp_WebSocket)]
+        public async Task FileUploadNotification_ReceiveMultipleNotificationsInOneConnection(TransportType transportType)
         {
-            fileUploaded = true;
-            return AcknowledgementType.Complete;
+            IotHubServiceClientOptions options = new IotHubServiceClientOptions()
+            {
+                UseWebSocketOnly = transportType == TransportType.Amqp_WebSocket,
+            };
+
+            using var serviceClient = new IotHubServiceClient(TestConfiguration.IoTHub.ConnectionString, options);
+
+            int fileUploadNotificationReceivedCount = 0;
+            Func<FileUploadNotification, AcknowledgementType> OnFileUploadNotificationReceived = (fileUploadNotification) =>
+            {
+                fileUploadNotificationReceivedCount++;
+                return AcknowledgementType.Complete;
+            };
+
+            serviceClient.FileUploadNotificationProcessor.FileUploadNotificationProcessor = OnFileUploadNotificationReceived;
+
+            await serviceClient.FileUploadNotificationProcessor.OpenAsync().ConfigureAwait(false);
+            await UploadFile().ConfigureAwait(false);
+            await UploadFile().ConfigureAwait(false);
+
+            // The open file upload notification processor should be able to receive more than one
+            // file upload notification without closing and re-opening as long as there is more
+            // than one file upload notification to consume.
+            WaitForFileUploadNotification(ref fileUploadNotificationReceivedCount, 2);
+            await serviceClient.FileUploadNotificationProcessor.CloseAsync().ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Wait until the expected number of file upload notifications have been received. If the expected
+        /// number of notifications are not received in time, this method throws a AssertionFailedException.
+        /// </summary>
+        /// <param name="fileUploadNotificationReceivedCount">The current number of file upload notifications received.</param>
+        /// <param name="expectedFileUploadNotificationReceivedCount">The expected number of file upload notifications to receive in this test.</param>
+        private void WaitForFileUploadNotification(ref int fileUploadNotificationReceivedCount, int expectedFileUploadNotificationReceivedCount)
+        {
+            var timer = Stopwatch.StartNew();
+            while (fileUploadNotificationReceivedCount < expectedFileUploadNotificationReceivedCount && timer.ElapsedMilliseconds < 60000)
+            {
+                Thread.Sleep(200);
+            }
+
+            timer.Stop();
+
+            if (fileUploadNotificationReceivedCount < expectedFileUploadNotificationReceivedCount)
+                throw new AssertionFailedException("Timed out waiting to receive file upload notification.");
```

Comment:
```suggestion
            fileUploadNotificationReceivedCount.Should().BeLessThan(expectedFileUploadNotificationReceivedCount, "Timed out waiting to receive file upload notification.");
```

Reply:
No, we want the received count of file upload notifications to be greater than or equal to the expected count, not the other way around

Reply:
I'll use fluent assertions though, sure

Reply:
My assertion was wrong, but you get the idea. Thanks.